cmd/containerd-shim-runhcs-v1/service.go

```diff
@@ -475,3 +522,16 @@ func (s *service) ComputeProcessorInfo(ctx context.Context, req *extendedtask.Co
 	r, e := s.computeProcessorInfoInternal(ctx, req)
 	return r, errdefs.ToGRPC(e)
 }
+
+func (s *service) Done() <-chan struct{} {
+	return s.shutdown
+}
+
+func (s *service) IsShutdown() bool {
+	select {
+	case <-s.shutdown:
+		return true
+	default:
+		return false
+	}
+}
```

Comment:
Nit: If this function is only required for the tests the maybe we should move it into one of the test files like so:

```
func IsServiceShutdown(s *service) bool {
...
}
```

Reply:
I was trying to future proof it, if some code needed to check if the service was in a finished state and it should cleanup.
It is completely unused right now, so I can remove it if you dont think it adds value.

Reply:
Fine to keep this too.